### DISCUSSION TITLE
Re-enabling the Guichan formula

### DISCUSSION THREAD

**Discussion by bjorn** [Homebrew/discussions#5130] [Writing Formulae/Casks]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.2.8-48-g24c8c0e
ORIGIN: https://github.com/Homebrew/brew
HEAD: 24c8c0e745017507d6bd34a9ccc9e703f437eb3c
Last commit: 4 hours ago
Core tap HEAD: 5033e0f3e1d977c0c0dd2d859467f1c965918775
Core tap last commit: 8 days ago
Core tap JSON: 14 Feb 12:14 UTC
Core cask tap JSON: 14 Feb 12:14 UTC
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_GITHUB_API_TOKEN: set
HOMEBREW_MAKE_JOBS: 12
HOMEBREW_SORBET_RUNTIME: set
Homebrew Ruby: 3.1.4 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/3.1.4/bin/ruby
CPU: dodeca-core 64-bit kabylake
Clang: 15.0.0 build 1500
Git: 2.39.3 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 8.1.2 => /usr/bin/curl
macOS: 14.1-x86_64
CLT: N/A
Xcode: 15.2
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  openssl@1.1

Warning: You have unlinked kegs in your Cellar.
Leaving kegs unlinked can lead to build-trouble and cause formulae that depend on
those kegs to fail to run properly once built. Run `brew link` on these:
  tcl-tk
```

### Description of issue

Guichan was deprecated in https://github.com/Homebrew/homebrew-core/pull/122334 and later disabled entirely in https://github.com/Homebrew/homebrew-core/pull/151272. I'd like to re-enable this formula because I'm trying to compile my application based on it. I'm also interested in resuming maintenance of Guichan myself, which now has its official repository at https://github.com/darkbitsorg/guichan.

I thought I'd start by forking guichan using the `bump-formula-pr` command, but it doesn't let me:

```
bjorn@mini4 ~ % brew bump-formula-pr guichan       
Error: This formula is disabled!
```

I guess I'll need to perform this change in a fork manually?

Of course, alternatively I could add Guichan as a git submodule to my repository.

**Comment by SMillerDev** [comment]
> I guess I'll need to perform this change in a fork manually?

Yes. However, unless the fork is deemed an official replacement, we won't be switching to the fork.

**Reply by bjorn** [reply]
It's not a fork though. This has been put online by one of the original developers, and I am also a contributor to Guichan.

**Reply by SMillerDev** [reply]
If I go to the current source location of the formula, will it redirect in any way to the new location?

**Reply by bjorn** [reply]
A link to the new repository has now been added to the original SourceForge page: https://sourceforge.net/projects/guichan/ (see blue box).

The formula currently downloads the code from code.google.com, but I don't think it's possible to redirect from there. At least the homepage is still listed as https://guichan.sourceforge.net/. Maybe it could still be made a forward.

**Comment by p-linnane** [comment]
You may want to host this [in your own Tap](https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap).

**Reply by bjorn** [reply]
Hmm, though in that case, I think using a git submodule will really be preferred. Otherwise, everybody who wants to build my game will need to set up another tap.

**Comment by bjorn** [comment]
After making a [Guichan 0.8.3 release](https://github.com/darkbitsorg/guichan/releases/tag/v0.8.3) I have resubmitted the Guichan formula at https://github.com/Homebrew/homebrew-core/pull/223673.
